temporian/core/event_set_ops.py

```diff
@@ -4436,3 +4437,115 @@ def assign(
         from temporian.core.operators.glue import assign
 
         return assign(self, **others)
+
+    def before(
+        self: EventSetOrNode,
+        timestamp: Union[int, float, datetime],
+    ) -> EventSetOrNode:
+        """Filter events [`EventSet`][temporian.EventSet] that happened before a
+        particual timestamp.
+
+        The timestamp can be a datetime if the EventSet's timestamps are unix
+        timestamps.
+
+        The comparison is strict, meaning that the obtained timestamps would be
+        less than (`<`) the provided timestamp.
+
+        Usage example:
+            ```python
+            >>> a = tp.event_set(
+            ...     timestamps=[0, 1, 5, 6],
+            ...     features={"f1": [0, 10, 50, 60]},
+            ... )
+
+            >>> a.before(5)
+            indexes: []
+            features: [('f1', int64)]
+            events:
+                 (2 events):
+                    timestamps: [0. 1.]
+                    'f1': [ 0 10]
+            ...
+
+            >>> from datetime import datetime
+            >>> a = tp.event_set(
+            ...     timestamps=[datetime(2022, 1, 1), datetime(2022, 1, 2)],
+            ...     features={"f1": [1, 2]},
+            ... )
+
+            >>> a.before(datetime(2022, 1, 1, 12))
+            indexes: []
+            features: [('f1', int64)]
+            events:
+                 (1 events):
+                    timestamps: ['2022-01-01T00:00:00']
+                    'f1': [1]
+            ...
+
+            ```
+
+        Args:
+            timestamp: EventSet with a single boolean feature.
```

Comment:
What about something like: Strict timestamp upper boundary (or something like that).